survivalstan/formulas.py

```diff
@@ -0,0 +1,315 @@
+                   stan_data=dict(), meta_data=dict(), **kwargs):
+        if patsy.util.have_pandas:
+            # prep pandas dataframe
+            dm = {'time': time,
+                 'event_status': event_status.astype(int),
+                 }
+            if group_id is not None:
+                dm.update({'group_id': group_id})
+            dm = pd.DataFrame(dm)
+            # prep stan_data object
+            stan_data.update({'y': dm['time'].values.astype(float),
+                              'event': dm['event_status'].values.astype(int),
+                              'N': len(dm.index)})
+            if group_id is not None:
+                stan_data.update({'g': dm['group_id'].values.astype(int)})
+            meta_data.update({'df': dm})
+        else:
+            raise ValueError('non-pandas usage not yet supported. Please',
+                             ' import pandas library to use `surv` syntax')
+            # prep np array
+            #if group_id is not None:
+            #    dm = np.append(time, event_status.astype(int), group_id, 1)
+            #else:
+            #    dm = np.append(time, event_status.astype(int), 1)
+            # prep stan_data object
+            stan_data.update({'y': time, 'event': event_status.astype(int), 'N': len(time)})
+            if group_id is not None:
+                stan_data.update({'g': group_id})
+        return SurvData(dm, stan_data=stan_data, meta_data=meta_data, **kwargs)
+
+    def transform(self, time, event_status, **kwargs):
+        kwargs = self._check_kwargs(**kwargs)
+        meta_data = dict()
+        stan_data = dict()
+        if 'subject' in kwargs.keys():
+            subject_id = self.subject_id.transform(kwargs['subject'])
+            timepoint_id = self.timepoint_id.transform(time)
+            meta_data.update({'timepoint_id': self.timepoint_id.decode_df(),
+                              'subject_id': self.subject_id.decode_df()})
+            stan_data.update(self._prep_timepoint_standata(self.timepoint_id.decode_df()))
+            stan_data.update({'S': self.subject_id.len()})
+        if 'group' in kwargs.keys():
+            group_id = self.group_id.transform(kwargs['group'])
+            stan_data.update({'G': self.group_id.len()})
+            meta_data.update({'group_id': self.group_id.decode_df()})
+        else:
+            group_id = None
+
+        if self._type == 'long':
+            return(self._prep_long(timepoint_id=timepoint_id, event_status=event_status,
+                                  subject_id=subject_id, group_id=group_id,
+                                  meta_data=meta_data, stan_data=stan_data)
+                  )
+        elif self._type == 'wide':
+            return(self._prep_wide(time=time, event_status=event_status, group_id=group_id,
+                                  meta_data=meta_data, stan_data=stan_data))
+
+surv = patsy.stateful_transform(Surv)
+
+def _get_args(s):
```

Comment:
Hi @armish - thanks for looking at this! 

I was surprised too - to my knowledge, patsy simply evals each Term in the parent environment & so doesn't actually parse the commands. I looked at python functions in `ast` & `parser` libraries but they didn't give me what I wanted (term names as strings) so ended up writing my own function. 

I will add a docstring & test cases for this. Agreed, it would be nice to modify this to use a more generic function for this use case.